packages/samples/headless-ssr-commerce/app/listing/page.tsx

```diff
@@ -1,15 +1,27 @@
+import {headers} from 'next/headers';
 import ListingPage from '../_components/listing-page';
-import {fetchStaticState} from '../_lib/commerce-engine';
+import {listingEngineDefinition} from '../_lib/commerce-engine';
+import {NextJsNavigatorContext} from '../_lib/navigatorContextProvider';
 
 /**
  * This file defines a List component that uses the Coveo Headless SSR commerce library to manage its state.
  *
  * The Listing function is the entry point for server-side rendering (SSR).
  */
 export default async function Listing() {
-  const staticState = await fetchStaticState();
+  // Sets the navigator context provider to use the newly created `navigatorContext` before fetching the app static state
+  const navigatorContext = new NextJsNavigatorContext(headers());
```

Comment:
I asked the same question in one of @alexprudhomme 's PR, so feel free to ignore if you've already answered there, but I wonder why we're doing this here instead of passing the navigator context provider when we define the engine.